client/hotrod-client/src/test/java/org/infinispan/client/hotrod/iteration/ProtobufRemoteIteratorTest.java

```diff
@@ -0,0 +1,119 @@
+package org.infinispan.client.hotrod.iteration;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.marshall.ProtoStreamMarshaller;
+import org.infinispan.client.hotrod.query.testdomain.protobuf.AccountPB;
+import org.infinispan.client.hotrod.query.testdomain.protobuf.marshallers.MarshallerRegistration;
+import org.infinispan.client.hotrod.test.MultiHotRodServersTest;
+import org.infinispan.commons.marshall.Marshaller;
+import org.infinispan.commons.util.Util;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.distribution.ch.ConsistentHash;
+import org.infinispan.filter.AbstractKeyValueFilterConverter;
+import org.infinispan.metadata.Metadata;
+import org.infinispan.query.remote.client.ProtobufMetadataManagerConstants;
+import static org.infinispan.server.hotrod.test.HotRodTestingUtil.hotRodCacheConfiguration;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+import static org.testng.AssertJUnit.assertFalse;
+import org.testng.annotations.Test;
+
+/**
+ * @author gustavonalle
+ * @since 8.0
+ */
+@Test(groups = "functional", testName = "client.hotrod.iteration.ProtobufRemoteIteratorTest")
+public class ProtobufRemoteIteratorTest extends MultiHotRodServersTest implements AbstractRemoteIteratorTest {
+
+   private static final int NUM_NODES = 2;
+   public static final int CACHE_SIZE = 10;
+
+   @Override
+   protected void createCacheManagers() throws Throwable {
+      ConfigurationBuilder cfgBuilder = hotRodCacheConfiguration(getDefaultClusteredCacheConfig(CacheMode.DIST_SYNC, false));
+      createHotRodServers(NUM_NODES, cfgBuilder);
+      waitForClusterToForm();
+
+      //initialize server-side serialization context
+      RemoteCache<String, String> metadataCache = client(0).getCache(ProtobufMetadataManagerConstants.PROTOBUF_METADATA_CACHE_NAME);
+      metadataCache.put("sample_bank_account/bank.proto", Util.read(Util.getResourceAsStream("/sample_bank_account/bank.proto", getClass().getClassLoader())));
+      assertFalse(metadataCache.containsKey(ProtobufMetadataManagerConstants.ERRORS_KEY_SUFFIX));
+
+      servers.forEach(s -> s.setFilterMarshallerClass(CustomProtoStreamMarshaller.class));
```

Comment:
Why do you need `setFilterMarshallerClass`? HotRodServer already has `setEventMarshaller`. Those two really should be the same thing, since the event marshaller is really used to unmarshaller filter/converter callback key/value instances